src/flint/fq_nmod.jl

```diff
@@ -480,6 +480,49 @@ rand(r::Random.AbstractRNG, R::FqNmodFiniteField, b::UnitRange{Int}) = rand(r, m
 
 rand(R::FqNmodFiniteField, b::UnitRange{Int}) = rand(Random.GLOBAL_RNG, R, b)
 
+###############################################################################
+#
+#   Iteration
+#
+###############################################################################
+
+Base.iterate(F::FqNmodFiniteField) = zero(F), zeros(UInt, degree(F))
+Base.iterate(F::FqFiniteField) = zero(F), zeros(fmpz, degree(F))
```

Comment:
Using `UInt` instead of `fmpz` leads to an iteration which is roughly twice as fast, but is limited to iterating over the first `typemax(UInt)` elements; if `UInt == UInt32` this might pose problem (a check could be made to error instead of giving wrong results). On the other hand, using `fmpz` allows to "jump forward" in the iteration by initializing the state manually (I don't know if someone would ever need this feature though). If the `fmpz` version is prefered, it's certainly possible to optimize it by modifying in-place the coefficients in the state.

Reply:
I don't have a strong opinion on this one. May be a good idea to hear what @wbhart thinks.